src/set.rs

```diff
@@ -107,5 +146,385 @@ where
+            assert!(set.insert(i).is_none());
+        }
+        for i in 0..len {
+            assert_eq!(i, set.get(&i).unwrap());
+            assert_eq!(i, set.remove(&i).unwrap());
+            assert!(set.get(&i).is_none());
+            set.raw.assert_pruned();
+        }
+
+        assert!(set.is_empty());
+    }
+
+    #[test]
+    fn remove_many() {
+        remove_many_inner(ConSet::new(), TEST_BATCH);
+    }
+
+    #[test]
+    fn remove_many_collision() {
+        remove_many_inner(ConSet::with_hasher(NoHasher), TEST_BATCH_SMALL);
+    }
+
+    #[test]
+    fn collision_remove_one_left() {
+        let mut set = ConSet::with_hasher(NoHasher);
+        set.insert(1);
+        set.insert(2);
+
+        set.raw.assert_pruned();
+
+        assert!(set.remove(&2).is_some());
+        set.raw.assert_pruned();
+
+        assert!(set.remove(&1).is_some());
+
+        set.raw.assert_pruned();
+        assert!(set.is_empty());
+    }
+
+    #[test]
+    fn collision_remove_one_left_with_str() {
+        let mut set = ConSet::with_hasher(NoHasher);
+        set.insert("hello");
+        set.insert("world");
+
+        set.raw.assert_pruned();
+
+        assert!(set.remove("world").is_some());
+        set.raw.assert_pruned();
+
+        assert!(set.remove("hello").is_some());
+
+        set.raw.assert_pruned();
+        assert!(set.is_empty());
+    }
+
+    #[test]
+    fn remove_par() {
+        let mut set = ConSet::new();
+        for i in 0..TEST_THREADS * TEST_BATCH {
+            set.insert(i);
+        }
+
+        thread::scope(|s| {
+            for t in 0..TEST_THREADS {
+                let set = &set;
+                s.spawn(move |_| {
+                    for i in 0..TEST_BATCH {
+                        let num = t * TEST_BATCH + i;
+                        let val = set.remove(&num).unwrap();
+                        assert_eq!(num, val);
+                        assert_eq!(num, val);
+                    }
+                });
+            }
+        })
+        .unwrap();
+
+        set.raw.assert_pruned();
+        assert!(set.is_empty());
+    }
+
+    fn iter_test_inner<S: BuildHasher>(set: ConSet<(usize, usize), S>) {
+        for i in 0..TEST_BATCH_SMALL {
+            assert!(set.insert((i, i)).is_none());
+        }
+
+        let mut extracted = set
+            .iter()
+            .map(|(i, n)| {
+                assert_eq!(i, n);
+                n
+            })
+            .collect::<Vec<_>>();
+
+        extracted.sort();
+        let expected = (0..TEST_BATCH_SMALL).into_iter().collect::<Vec<_>>();
+        assert_eq!(expected, extracted);
+    }
+
+    #[test]
+    fn iter() {
+        let set = ConSet::new();
+        iter_test_inner(set);
+    }
+
+    #[test]
+    fn iter_collision() {
+        let set = ConSet::with_hasher(NoHasher);
+        iter_test_inner(set);
+    }
+
+    #[test]
+    fn collect() {
+        let set = (0..TEST_BATCH_SMALL)
+            .into_iter()
+            .map(|i| (i, i))
```

Comment:
Again, is this just leftover from the map tests? I think this line could simply be omitted (and the map few lines below too) and it would still make the same sense.